File: javascriptp/OOPS/js_coupling/book.js
//Strong Coupling
var Book = function (name, cost, author) {
  this.name = name;
  this.cost = cost;
  this.author = author;
};

var Knowledge = function (name, cost, author, pages) {
  Book.call(this, name, cost, author);
  this.pages = pages;
};

var k1 = new Knowledge("Blue Ocean", 20, "Avinash Nishad", 20);
console.log("k1", k1.author);

//Weak Coupling
function Vehicle() {
  this.vehicleName;
  throw new Error("You cannot create an instance of Abstract Class");
}

Vehicle.prototype.display = function () {
  return "Vehicle is: " + this.vehicleName;
};

//Creating a constructor function
function Bike(vehicleName) {
  this.vehicleName = vehicleName;
}

//Creating object without using the function constructor
Bike.prototype = Object.create(Vehicle.prototype);
var bike = new Bike("Honda");
console.log("bike", bike);
console.log("bike", bike.display());

File: pythonp/OOPS/py_coupling/book.py
from abc import ABC


#Strong coupling
class Book:
    def __init__(self, name):
        self.name = name

    def showName(self):
        print(self.name)


class Blue(Book):
    pass


print("Strong coupling")
b1 = Blue("Book")
b1.showName()

#Weak coupling


class Person(ABC):
    def showName(self):
        pass


class Teacher(Person):
    def showName(self):
        print("Teacher Person")


print("Weak coupling")
t1 = Teacher()
t1.showName()